### ISSUE TITLE
library.deploy ignoring -options json

### ISSUE THREAD

**Issue by serokles** [vmware/govmomi#1718]
I am trying to provision an OVF from a subscribed content library using govc.  I can use library.deploy to create vms from the library item but I am trying to use the -options flag to inject a json file including all the properties I need.  I have a feeling that the -options flag is being ignored because the following commands have the following result:

```
govc library.deploy /vGhettoLib/Nested_ESXi6.7u3_Appliance_Template_v1.0 -options vesxitemp.json
```
This generates a VM called "-options".  I thought perhaps the "=" needed t be included to mimic the command for import.ova:
```
govc library.deploy /vGhettoLib/Nested_ESXi6.7u3_Appliance_Template_v1.0 -options=vesxitemp.json
```
But this generates a VM called "-options=vesxitemp.json".  I decided to try using the VM Name as well to see if that could work around the issue:
```
govc library.deploy /vGhettoLib/Nested_ESXi6.7u3_Appliance_Template_v1.0 vesxi4 -options=vesxitemp.json
```
But this generates a correctly named VM but with no OVF spec injected meaning the vm does not have the correct IP information, is not powered on etc etc.  The spec I am using was generated by downloading the OVF from the library and using import.spec to generate a json file I could then place my values in.  The file is below:

```
{
  "DiskProvisioning": "flat",
  "IPAllocationPolicy": "fixedPolicy",
  "IPProtocol": "IPv4",
  "PropertyMapping": [
    {
      "Key": "guestinfo.hostname",
      "Value": "vesxi4"
    },
    {
      "Key": "guestinfo.ipaddress",
      "Value": "10.64.0.24"
    },
    {
      "Key": "guestinfo.netmask",
      "Value": "255.255.255.0"
    },
    {
      "Key": "guestinfo.gateway",
      "Value": "10.64.0.1"
    },
    {
      "Key": "guestinfo.vlan",
      "Value": ""
    },
    {
      "Key": "guestinfo.dns",
      "Value": "192.168.1.44"
    },
    {
      "Key": "guestinfo.domain",
      "Value": "houseofbears.co.uk"
    },
    {
      "Key": "guestinfo.ntp",
      "Value": "pool.ntp.org"
    },
    {
      "Key": "guestinfo.syslog",
      "Value": "192.168.1.200"
    },
    {
      "Key": "guestinfo.password",
      "Value": "VMware1!"
    },
    {
      "Key": "guestinfo.ssh",
      "Value": "True"
    },
    {
      "Key": "guestinfo.createvmfs",
      "Value": "False"
    }
  ],
  "NetworkMapping": [
    {
      "Name": "pg_LabMgmt",
      "Network": "pg_LabMgmt"
    }
  ],
  "Annotation": "Nested ESXi 6.7 Update 3 Appliance (Build 14320388) www.virtuallyghetto.com",
  "MarkAsTemplate": false,
  "PowerOn": true,
  "InjectOvfEnv": true,
  "WaitForIP": false,
  "Name": "vesxi4"
}

```

Does the OVF JSON file need any other special headers or element names added to be imported by the deploy := vcenter.Deploy part of library.deploy?

**Comment by dougm** [comment]
Due to how Go's flag parsing works, the `-options foo.json` has to come before the arguments:
```
 govc library.deploy -options=vesxitemp.json /vGhettoLib/Nested_ESXi6.7u3_Appliance_Template_v1.0 vesxi4
```

**Comment by serokles** [comment]
Thanks for that @dougm 
Alas although I can now get it to read the file the contents of the additional parameters do not seem to end up being passed through to the VM.  Annotations, name and switches make it through but everything to do with guestinfo never makes it to the VM.  Looking at the debug I can see the parameters are in the req json but the contents of the syslog on the subsequent esxi VM have blank entries for everything.  Most baffling.

**Comment by nanw1103** [comment]
Similar to #1720 
To me it looks like a govc bug, a total blocker for production-related deployment scenario, which has, very likely, OVA properties involved.

**Comment by serokles** [comment]
Yeah I have moved to native API for this part as I can see it is not passing the additional_parameters in the request JSON and thus the cls.log shows no additional settings being passed and creates the VM from defaults.  When I get back to my laptop will try and dig out some log snippets to illustrate and if I ever learn go might have a crack at it!

**Comment by dougm** [comment]
Fixed by #1755 , let us know if that also works for you @serokles @nanw1103
